File: frontend/src/types/hotel.ts
export interface Address {
  street: string;
  city: string;
  state: string;
  zip_code: string;
  country: string;
}

export interface Contact {
  phone: string;
  email: string;
}

export interface Policy {
  check_in_time: string;
  check_out_time: string;
  cancellation: string;
}

export interface Room {
  room_id: string;
  type: string;
  bed_type: string;
  bed_count: number;
  max_occupancy: number;
  square_footage: number;
  price_per_night: number;
  room_amenities: string[];
  available_dates: string[];
}

export interface Hotel {
  id: string;
  name: string;
  description: string;
  star_rating: number;
  overall_rating: number;
  review_count: number;
  address: Address;
  contact: Contact;
  amenities: string[];
  policies: Policy;
  rooms: Room[];
}

export type SortOption =
  | ""
  | "name_asc"
  | "stars_desc"
  | "rating_desc"
  | "price_asc"
  | "price_desc";

export interface FilterState {
  city: string;
  minStars: number;
  minPrice: number;
  maxPrice: number;
  sortBy: SortOption;
}

File: backend/internal/models/hotel.go
package models

// Address is a physical location.
type Address struct {
	Street  string `json:"street"`
	City    string `json:"city"`
	State   string `json:"state"`
	ZipCode string `json:"zip_code"`
	Country string `json:"country"`
}

// Contact holds phone and email for a property.
type Contact struct {
	Phone string `json:"phone"`
	Email string `json:"email"`
}

// Policy describes check-in/out and cancellation terms.
type Policy struct {
	CheckInTime  string `json:"check_in_time"`
	CheckOutTime string `json:"check_out_time"`
	Cancellation string `json:"cancellation"`
}

// Room is a bookable room type within a hotel.
type Room struct {
	RoomID         string   `json:"room_id"`
	Type           string   `json:"type"`
	BedType        string   `json:"bed_type"`
	BedCount       int      `json:"bed_count"`
	MaxOccupancy   int      `json:"max_occupancy"`
	SquareFootage  int      `json:"square_footage"`
	PricePerNight  float64  `json:"price_per_night"`
	RoomAmenities  []string `json:"room_amenities"`
	AvailableDates []string `json:"available_dates"`
}

// Hotel is a full property record.
type Hotel struct {
	ID            string   `json:"id"`
	Name          string   `json:"name"`
	Description   string   `json:"description"`
	StarRating    int      `json:"star_rating"`
	OverallRating float64  `json:"overall_rating"`
	ReviewCount   int      `json:"review_count"`
	Address       Address  `json:"address"`
	Contact       Contact  `json:"contact"`
	Amenities     []string `json:"amenities"`
	Policies      Policy   `json:"policies"`
	Rooms         []Room   `json:"rooms"`
}

// HotelFilter holds optional search criteria for SearchHotels.
type HotelFilter struct {
	City     string
	MinStars int
	MaxStars int
	MinPrice float64
	MaxPrice float64
}
